Classes/common/touchdb/TDBlobStore.m

```diff
@@ -187,6 +190,8 @@ - (NSInteger)deleteBlobsExceptWithKeys:(NSSet*)keysToKeep withDatabase:(FMDataba
             [NSData dataWithBytes:oneRow.key.bytes length:sizeof(oneRow.key.bytes)];
         if ([keysToKeep containsObject:curKeyData]) {
             // Do not delete blob. It is an exception.
+            [filesToKeep addObject:oneRow.blobFilename];
+
             continue;
         }
 
```

Comment:
You are right, `numDeleted` is only used in `garbageCollectAttachments` but also notice that if we are able to delete the file from disk, we [delete the row from db](https://github.com/cloudant/CDTDatastore/pull/170/files#diff-ce77b2812a143c42aa7fdb8b3818581dR205).

Probably `deleteFilesExceptWithFilenames:` is not a good name or it does not explain completely what it does: it deletes the files in disk that are not related to an attachment in db.